db/attachment.go

```diff
@@ -218,29 +236,37 @@ type AttachmentCallback func(name string, digest string, knownData []byte, meta
 // its data. The callback is told whether the attachment body is known to the database, according
 // to its digest. If the attachment isn't known, the callback can return data for it, which will
 // be added to the metadata as a "data" property.
-func (db *Database) ForEachStubAttachment(body Body, minRevpos int, callback AttachmentCallback) error {
+func (db *Database) ForEachStubAttachment(body Body, minRevpos int, docID string, callback AttachmentCallback) error {
 	atts := GetBodyAttachments(body)
 	if atts == nil && body[BodyAttachments] != nil {
-		return base.HTTPErrorf(400, "Invalid _attachments")
+		return base.HTTPErrorf(http.StatusBadRequest, "Invalid _attachments")
 	}
 	for name, value := range atts {
 		meta, ok := value.(map[string]interface{})
 		if !ok {
-			return base.HTTPErrorf(400, "Invalid attachment")
+			return base.HTTPErrorf(http.StatusBadRequest, "Invalid attachment")
 		}
 		if meta["data"] == nil {
 			if revpos, ok := base.ToInt64(meta["revpos"]); revpos < int64(minRevpos) || !ok {
 				continue
 			}
 			digest, ok := meta["digest"].(string)
 			if !ok {
-				return base.HTTPErrorf(400, "Invalid attachment")
+				return base.HTTPErrorf(http.StatusBadRequest, "Invalid attachment")
 			}
-			data, err := db.GetAttachment(AttachmentKey(digest))
-			if err != nil && !base.IsDocNotFoundError(err) {
-				return err
+			version, ok := GetAttachmentVersion(meta)
+			if !ok {
+				return ErrAttachmentVersion
+			}
+			var data []byte
+			var err error
+			if version == AttVersion1 {
```

Comment:
Yeah, the issue was that the ["data"] getting populated in the attachment meta because of the getAttachment call and the actual data was not getting removed from the meta.